plugin/storage/es/factory.go

```diff
@@ -236,8 +256,57 @@ var _ io.Closer = (*Factory)(nil)
 
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
+	var errs []error
+	for _, w := range f.watchers {
+		errs = append(errs, w.Close())
+	}
 	if cfg := f.Options.Get(archiveNamespace); cfg != nil {
 		cfg.TLS.Close()
 	}
-	return f.Options.GetPrimary().TLS.Close()
+	errs = append(errs, f.Options.GetPrimary().TLS.Close())
+	return errors.Join(errs...)
+}
+
+// initializeClient initializes either primary or archive ES client
+func (f *Factory) initializeClient(config config.ClientBuilder, options []elastic.ClientOptionFunc,
```

Comment:
assuming this function is necessary, its interface should be simpler: create and return a client or error. There is no need to pass atomic holder, for example.